### ISSUE TITLE
Upgrading to php_codesniffer:2.x

### ISSUE THREAD

**Issue by ricog** [loadsys/loadsys_codesniffer#16]
I thought I would look into updating the codesniffer to php_codesniffer:2.x in order to take advantage of changing some of the issues in serenity. Is there value in pulling in recent cakephp-codesniffer changes? Here is a diff since the [Upstream rollup](https://github.com/loadsys/loadsys_codesniffer/commit/7dc38d62af23b64635ac3b9b8151c1ec59b63310) change @beporter made back in November.

https://github.com/cakephp/cakephp-codesniffer/compare/4c96a24610e831c58478d14a143157a057f655d9...master?w=1

**Comment by beporter** [comment]
Isn't a very large chunk of this going to be PSR-2 stuff? It's a huge diff, which worries me.

**Comment by beporter** [comment]
I can see there are a few things of value in there, but picking them out from the "opening brace on a new line" stuff might be challenging.

**Comment by jtyost2** [comment]
Yeah, a lot of it is the PSR-2 stuff, it's on my list to look into.

**Comment by ricog** [comment]
Thanks. Do you think there an easy route of getting what we already have working with phpcs 2.x?

**Comment by beporter** [comment]
I haven't been able to find any documentation about what changed in phpcs v2. I don't even know how the sniffs themselves have to change to be compatible. The information for the project doesn't seem to be well-consolidated. The release blog post is the most helpful thing I could find: http://www.squizlabs.com/php-codesniffer/2.0.0-released

**Comment by ricog** [comment]
We started tackling this in #17 which is currently just [pulling in cakephp and disabling a few rules](https://github.com/loadsys/loadsys_codesniffer/blob/php_codesniffer-2.x/Loadsys/ruleset.xml#L9-22).

I'll keep a list of tasks we are aware of that need fixed or confirmed in #17. For now the only thing that's really left of the old Loadsys sniffer is the private/protected name sniffs (although I haven't tested them). I did not copy the function comment logic because it was not obvious how to apply it, and I haven't yet confirmed if it's still needed.
